packages/lodestar-beacon-state-transition/src/fast/util/pubkeys.ts

```diff
@@ -0,0 +1,26 @@
+import {PublicKey} from "@chainsafe/bls";
+import {BeaconState, ValidatorIndex} from "@chainsafe/lodestar-types";
+import {ByteVector, toHexString} from "@chainsafe/ssz";
+
+export class PubkeyIndexMap extends Map<ByteVector, ValidatorIndex> {
+  get(key: ByteVector): ValidatorIndex | undefined {
+    return super.get(toHexString(key) as unknown as ByteVector);
+  }
+  set(key: ByteVector, value: ValidatorIndex): this {
+    return super.set(toHexString(key) as unknown as ByteVector, value);
+  }
+}
+
+export function syncPubkeys(pubkey2index: PubkeyIndexMap, index2pubkey: PublicKey[], state: BeaconState): void {
+  const currentCount = pubkey2index.size;
+  if (currentCount !== index2pubkey.length) {
+    throw new Error("Pubkey indices have fallen out of sync");
+  }
+  const newCount = state.validators.length;
+  for (let i = currentCount; i < newCount; i++) {
+    const pubkey = state.validators[i].pubkey.valueOf() as Uint8Array;
+    pubkey2index.set(pubkey, i);
+    index2pubkey.push(PublicKey.fromBytes(pubkey));
+  }
+}
+
```

Comment:
Since some code was shared between `EpochContext` instance methods and static methods, I pulled the logic into an external utility for easier reusing. Let me know if it's best to have this utility as a static method of `EpochContext`. Since no-one is using this utility outside of `EpochContext` it seemed cleaner to pull it out.